### PR TITLE
test: resourceaccess integration testing

### PR DESCRIPTION
**Pull request by chenbishop**
## Related Tickets & Documents

- Issue:
Adding integration test for resource access flow

## Changes
Adding integration test for resource access flow

## Notes
1. Added environmental variable "TWINGATE_TEST_PRINCIPAL_ID" as requirement for testing as resourceAccess requires the value
2. Added environtal variable "TWINGATE_REMOTE_NETWORK_ID" as requirement for testing as this is required by the operator

### CODE REVIEW

File: tests_integration/test_main.py [hunk 1]
```diff
@@ -542,3 +545,73 @@ def test_protocols_udp_restricted_port_values_start_must_be_lte_end(
...
+        spec:
+          resourceRef:
+            name: {unique_resource_name}
+          principalId: {principal_id}
+    """
+
+    # fmt: off
+    with KopfRunner(kopf_runner_args, settings=kopf_settings) as runner:
+        kubectl_create(RESOURCE_OBJ)
+        time.sleep(10)  # give it some time to react
```
Comment (ekampf) [inline tests_integration/test_main.py:580]: I dont think we need 2 sleeps, we can call `kubectl_create` twice then sleep for `5s`

File: tests_integration/test_main.py [hunk 2]
```diff
@@ -542,3 +545,73 @@ def test_protocols_udp_restricted_port_values_start_must_be_lte_end(
...
+        resource_delete_command = kubectl(f"delete tgr/{unique_resource_name}")
+
+    # fmt: on
+
+    # Ensure that the operator did not die on start, or during the operation.
+    assert runner.exception is None
+    assert runner.exit_code == 0
+
+    logs = load_stdout(runner.stdout)
+    for log in logs:
```
Comment (ekampf) [inline tests_integration/test_main.py:600]: This is a debug leftover - should delete it from `test_resource_flows` too

File: tests_integration/test_main.py [hunk 3]
```diff
@@ -542,3 +545,73 @@ def test_protocols_udp_restricted_port_values_start_must_be_lte_end(
 
         stderr = ex.value.stderr.decode()
         assert "Start port value must be less or equal to end port value" in stderr
+
+
+@pytest.mark.integration()
+def test_resource_access_flows(kopf_settings, unique_resource_name):
+    assert "TWINGATE_PRINCIPAL_ID" in os.environ
```
Comment (ekampf) [inline tests_integration/test_main.py:552]: Lets rename `TWINGATE_PRINCIPAL_ID` to `TWINGATE_TEST_PRINCIPAL_ID` so we know this argument is used for tests

File: tests_integration/test_main.py (hunk 1 above)
Comment (chenbishop) [reply tests_integration/test_main.py:580]: I tried with 5s sleep but the resourceAccess creating was failing (error was the resource not created yet)

File: tests_integration/test_main.py (hunk 3 above)
Comment (ekampf) [reply tests_integration/test_main.py:552]: I added a valid `TWINGATE_TEST_PRINCIPAL_ID` to CI so your PR can pass tests

Comment (chenbishop) [reply tests_integration/test_main.py]: recommended change applied

File: tests_integration/test_main.py (hunk 2 above)
Comment (chenbishop) [reply tests_integration/test_main.py]: recommended change applied

**Review by ekampf — APPROVED** [review]

**Pull request merged**